rust/vectordb/src/io/object_store.rs

```diff
@@ -118,8 +118,10 @@ impl ObjectStore for MirroringObjectStore {
         self.primary.head(location).await
     }
 
-    // garbage collection on secondary will happen async from other means
     async fn delete(&self, location: &Path) -> Result<()> {
+        if !location.primary_only() {
+            self.secondary.delete(location).await?;
```

Comment:
if the secondary delete fails then we also don't do the primary delete? is that intentional?

Reply:
delete is usually just for cleaning up tmp files or garbage collect. This is okay if we fail on secondary